engine/src/main/java/nl/inl/blacklab/searches/Search.java

```diff
@@ -19,20 +17,87 @@
      *
      * Executes the search synchronously.
      *
+     * This search may be queued if load is high. To avoid this (and potential
+     * deadlock situations because of it), use {@link #executeNoQueue()} or {@link #execute(boolean)}.
+     *
+     * @return result of the operation
+     * @throws InvalidQuery
+     */
+    default R execute() throws InvalidQuery {
```

Comment:
Yes, but I feel this way is more readable:
- I feel `Search.execute()` is much more obvious than having to understand an extra boolean parameter, especially for new users.
- I prefer `execute()` to `executeSync()` because the default for methods is that they execute synchronously (we don't have `.toStringSync()` either). `execute / executeAsync` are also clearly different whereas `executeSync / executeAsync` could be mistaken for the same name if you're not reading carefully.
- Again, I think `executeNoQueue` makes it clear to someone reading the code that this `Search` will not be queued, whereas `execute(false)` requires you to look up what the boolean does if you don't know or remember. I would probably be in favour of doing this more. It's unfortunate that Java doesn't support named parameters, e.g. `execute(allowQueue=false)`.

Reply:
Alright, it was a very nitpicky comment anyway ;) Named and default parameters would be great, but we'll have to make do with what we have. 
I guess technically one could use an enum parameter to emulate it, but that feels so much worse that just having couple differently named functions, so let's not..